src/Microsoft.Azure.SignalR/Internal/ProductInfo.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.Azure.SignalR
+{
+    internal static class ProductInfo
+    {
+        public static string GetProductInfo()
+        {
+            var packageId = typeof(ProductInfo).GetTypeInfo().Assembly.GetName().Name;
+            var version = typeof(ProductInfo).GetTypeInfo().Assembly.GetCustomAttribute(typeof(AssemblyInformationalVersionAttribute)) as AssemblyInformationalVersionAttribute;
```

Comment:
>GetCustomAttribute [](start = 69, length = 18)

can use the generic version `GetCustomAttribute<AssemblyInformationalVersionAttribute>()`

Reply:
cool.